lib/SILOptimizer/Mandatory/Differentiation.cpp

```diff
@@ -2314,18 +2311,12 @@ class PrimalGenCloner final
       auto *proto = conformance.getAbstract();
       assert(proto && "Expected protocol in generic signature requirement");
       auto reqType = req.getFirstType();
-      // If requirement type can be substituted in original substutition map to
-      // form a non-archetype type, use the substituted type.
-      if (auto origFirstType = reqType.subst(origSubstMap))
-        if (!origFirstType->hasArchetype())
-          reqType = origFirstType;
-      // If requirement type has no type parameters and is not an archetype type,
-      // it is valid. Continue.
+      // If requirement type has no type parameters and is not an archetype
+      // type, it is valid. Continue.
       if (!reqType->isTypeParameter() && !reqType->hasArchetype())
```

Comment:
hasArchetype() will always be false here

Reply:
Logic revamped in https://github.com/apple/swift/pull/22243/commits/0fb115dcc725e543b78d4994ace58eabf8ed69cf.